shadowsocks/tcp.go

```diff
@@ -71,39 +54,49 @@ func debugTCP(cipherID, template string, val interface{}) {
 	}
 }
 
-func findAccessKey(clientConn onet.DuplexConn, cipherList CipherList) (string, onet.DuplexConn, []byte, error) {
-	// This must have enough space to hold the salt + 2 bytes chunk length + AEAD tag (Overhead) for any cipher
-	firstBytes := make([]byte, 0, 32+2+16)
-	// Constant of zeroes to use as the start chunk count. This must be as big as the max NonceSize() across all ciphers.
-	zeroCountBuf := [12]byte{} // MaxCountSize
+func findAccessKey(clientConn onet.DuplexConn, cipherList CipherList, m metrics.ShadowsocksMetrics) (string, onet.DuplexConn, []byte, error) {
```

Comment:
Let's return the duration instead of depending on ShadowsocksMetrics. That's quite a high-level (application-level) concept for this low level function.

Reply:
Done